util/network.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util // import "istio.io/fortio/util"
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+)
+
+const (
+	// MinTCPPort is the minimum TCP port number that can be used by Fortio http server.
+	MinTCPPort int = 1024
+	// MaxTCPPort is the maximum TCP port number that can be used by Fortio http server.
+	MaxTCPPort int = 65535
+)
+
+// NormalizePort validates hostport and returns hostport as :port or host:port.
+// If the host:port or port are invalid, NormailzePort returns an error.
+func NormalizePort(hostport string) (string, error) {
```

Comment:
we only need to see if we need to add just a : in front of the service, for the default to be like today

I'm aware ipv6 have : in them and the standard way to deal with that is [ipv6]:port which I expect (but I admit I haven't tested with go listener) just works ? (just need to add a test using [::1]:port for localhost

so the logic I think is really just : does the string contains any : if not use :string

Reply:
Understood. Adding 0.0.0.0 serves the following purposes and provides the same default behavior:

 - It is explicit. 0.0.0.0 tells the user that the service is listening on all IP's.
 - It's a clean solution for providing the UI URL to the user.

Reply:
What if a user provides `:8080` instead of `8080`? I think it's a good idea to use `SplitHostPort` from the net pkg to validate. However, I will defer to you if you have a strong opinion on the subject.

Reply:
you can probably extract the bind address from the listener after it is bound as opposed to try to make it on the fortio side

it would solve maybe the other "todo" which is it'd be great to be able to use "0" as the port and get the actual port

Reply:
as I said; either there is already a ":" in the string and we then use it as is, or there isn't and we add it - it's very simple 3 line if

Reply:
Got it. I'll resubmit the PR based on your feedback. Thank you.